astropy/table/table.py

```diff
@@ -264,7 +269,13 @@ def __init__(self, data=None, masked=None, names=None, dtype=None,
                 default_names = data.dtype.names
             else:
                 init_func = self._init_from_ndarray  # _homog
-                n_cols = data.shape[1]
+                if data.shape == ():
+                    raise ValueError('Can not initialize a Table with a scalar')
+                elif len(data.shape) == 1:
+                    n_cols = data.shape[0]
+                    data = np.expand_dims(data, axis=0)
```

Comment:
Totally trivial, and only if you haven't done the last change yet, but writing this as
```
data = data[np.newaxis, :]
```
or (bit less good)
```
data = data.reshape(1, -1)
```
is clearer to me at least.

(Actually, if you do this before the `n_cols` assignment, you can remove the `else:` clause.)